onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc

```diff
@@ -515,7 +522,34 @@ Status ModelBuilder::Compile(std::unique_ptr<Model>& model) {
       nnapi_->ANeuralNetworksModel_finish(nnapi_model_->model_),
       "on model finish");
 
+  // We have a list of target devices, try to see if the model can be run entirely
+  // using the list of target devices
+  bool use_create_for_devices = false;
   if (!nnapi_target_devices_.empty()) {
+    bool supported_ops[num_nnapi_ops_];
```

Comment:
i think that the array needs to have a constant value as the dimension. interestingly, it seems only MSVC fails to compile something like this. https://godbolt.org/z/G96c3Y
maybe use std::unique_ptr<bool[]> instead?

Reply:
I didn't know that, changed to use std::unique_ptr<bool[]>

Reply:
cool, would also suggest using onnxruntime::make_unique<bool[]>(num_nnapi_ops_)